src/generators.rs

```diff
@@ -1912,6 +1912,28 @@ pub fn gen_analysis_pass_type_checker(input_size: u16) -> (Option<String>, Strin
     (Some(setup_body), body)
 }
 
+/// Returns tuple of optional setup clarity code, and "main" clarity code
+pub fn gen_analysis_pass(function: AnalysisCostFunction, _scale: u16, input_size: u16) -> (Option<String>, String) {
+    match function {
+        AnalysisCostFunction::ReadOnly => gen_analysis_pass_read_only(input_size),
+        AnalysisCostFunction::TypeChecker => gen_analysis_pass_type_checker(input_size),
+        AnalysisCostFunction::TraitChecker => gen_analysis_pass_trait_checker(input_size),
+        AnalysisCostFunction::ArithmeticOnlyChecker => gen_analysis_pass_arithmetic_only(input_size),
+    }
+}
+
+// contract-call-bench? does everything contract-call? does, except load and execute the contract code
+pub fn gen_contract_call(scale: u16) -> (Option<String>, String) {
+    // let mut setup_body = String::new();
+    let mut body = String::new();
+
+    for _ in 0..scale {
+        body.push_str("(contract-call-bench? 'SP000000000000000000002Q6VF78.cost-voting get-counter) ");
```

Comment:
Not really sure how many args should be supplied here...

Reply:
hmm ya I would think it doesn't matter too much as well